src/twister2/quarantine_plugin.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import pytest
+import logging
+
+from pathlib import Path
+from yaml import safe_load
+from marshmallow import Schema, fields, ValidationError
+from dataclasses import dataclass, field
+
+from twister2.report.helper import get_test_name, get_item_platform
+from twister2.exceptions import TwisterConfigurationException
+
+logger = logging.getLogger(__name__)
+
+
+class QuarantinePlugin:
+    """Handle tests under quarantine"""
+
+    def __init__(self, config: pytest.Config) -> None:
+        """
+        :param config: pytest configuration
+        """
+        self.quarantine_verify = True if 'quarantine' in config.getoption('-m') else False
+        self.quarantine = QuarantineData()
+        for quarantine_file in config.getoption('--quarantine-list'):
+            self.quarantine.extend(QuarantineData.load_data_from_yaml(quarantine_file))
+
+    def pytest_collection_modifyitems(
+        self, session: pytest.Session, config: pytest.Config, items: list[pytest.Item]
+    ):
+        for item in items:
+            if qelem := self.quarantine.get_matched_quarantine(item):
+                item.add_marker(pytest.mark.quarantine(qelem.comment))
+                # do not mark tests to be skipped if want to verify quarantined tests
+                if not self.quarantine_verify:
+                    item.add_marker(pytest.mark.skip('under quarantine'))
+                    logger.debug("Skipped test %s - quarantine reason: %s" % (get_test_name(item), qelem.comment))
+
+
+@dataclass
+class QuarantineElem:
```

Comment:
We should not use `abbreviations` in class name. `QuarantineElement` is better name.